src/megaapi_impl.cpp

```diff
@@ -23107,6 +23148,128 @@ void MegaApiImpl::sendPendingRequests()
+            const string* handleContainerStr = u->getattr(ATTR_MY_BACKUPS_FOLDER);
+            if (!handleContainerStr) { e = API_EACCESS; break; }
+
+            std::unique_ptr<TLVstore> tlvRecords(TLVstore::containerToTLVrecords(handleContainerStr, &client->key));
+            if (!tlvRecords || !tlvRecords->find("h")) { e = API_ENOENT; break; }
+
+            const string& handleStr = tlvRecords->get("h");
+            handle h = MegaApi::base64ToHandle(handleStr.c_str());
+            if (h == UNDEF) { e = API_ENOENT; break; }
+
+            // get Node of remote "My Backups" folder
+            MegaNode* myBackupsNode = getNodeByHandle(h);
+            if (!myBackupsNode) { e = API_ENOENT; break; }
+
+            // get `DEVICE_NAME`, from user attributes
+            if (!u->isattrvalid(ATTR_DEVICE_NAMES)) { e = API_EINCOMPLETE; break; }
+            const string* deviceNameContainerStr = u->getattr(ATTR_DEVICE_NAMES);
+            if (!deviceNameContainerStr) { e = API_EINCOMPLETE; break; }
+
+            tlvRecords.reset(TLVstore::containerToTLVrecords(deviceNameContainerStr, &client->key));
+            if (!tlvRecords || !tlvRecords->find("n")) { e = API_EINCOMPLETE; break; } // is 'n' the right key ??
+
+            const string& devName64Str = tlvRecords->get("n");
+            string deviceName = Base64::btoa(devName64Str);
+            if (deviceName.empty()) { e = API_EINCOMPLETE; break; }
+
+            // get 'device-id'
+            const string& deviceId = client->getDeviceid();
+            if (deviceId.empty()) { e = API_EINCOMPLETE; break; }
+
+            // loop remote children of "My Backups" folder, to find `DEVICE_NAME`-with-'device-id' node
+            const MegaNodeList* nodeList = myBackupsNode->getChildren();
+            MegaNode* deviceNameNode = nullptr;
+            for (int i = 0; i < nodeList->size(); ++i)
+            {
+                MegaNode* child = nodeList->get(i);
+
+                // get folder name
+                const char* name = child->getName();
+                bool gotName = deviceName == name;
+
+                // get 'device-id' tag
+                const char* id = child->getCustomAttr("device-id");
+                bool gotId = deviceId == id;
+
+                // determine whether we found our node or not
+                if (!gotName && !gotId) { continue; } // ignore this child folder
+
+                if (gotName && gotId) // got it, use this one to add another backup to it
+                {
+                    deviceNameNode = child;
+                    break;
+                }
+
+                // there's an error if only one of the folder's {name, id} is correct and the other is wrong.
+                e = API_EINCOMPLETE;
+                break;
+            }
+
+            if (e != API_OK) { break; }
+
+            if (!deviceNameNode)
+            {
+                // create "My Backups"/`DEVICE_NAME`/
+                // HOW ? Like this?
```

Comment:
You just need to create the folder, not a sync. It should be something similar to the code for `TYPE_CREATE_FOLDER`. Additionally, the node's attribute should be specified. Once the folder is created, you can continue to create the subfolder for the backup itself. There's a possible way to create subtrees at once (so you could create the `DEVICE_NAME` folder and the backup folder at one shot), which could be better.

Reply:
Do you mean something like the code in
```
MegaApiImpl::sendPendingRequests()

    case MegaRequest::TYPE_CREATE_FOLDER:
```
?

Reply:
Yes, exactly. Perhaps not every line of code in that `case` makes sense for this use case, so you can probably simplify it. Feel free to create new methods to reuse common parts, if suitable.